src/account-info/AccountDetails.js

```diff
@@ -51,6 +51,11 @@ export default class AccountDetails extends PureComponent<Props, void> {
           <RawLabel style={[styles.largerText, styles.halfMarginLeft]} text={user.email} />
         </View>
         <View style={[styles.row, styles.margin, styles.center]}>
+          {/* Flow complains because ActivityText has a `presence` prop which
+            we do not specify here. However, we don't need to, since the
+            `presence` prop gets automatically assigned in ActivityText's
+            creation with `connect`.
+            $FlowFixMe */}
```

Comment:
You can mark the `presences` as optional ala:
```js
presences?: PresenceState,
```

Or maybe there is a better way to tell Flow the component is always getting the param from `connect`?
I think we should use $FlowFixMe sparingly, and possibly remove many of the ones we already have.

Reply:
> I think we should use $FlowFixMe sparingly, and possibly remove many of the ones we already have.

I agree. This series of commits doesn't actually introduce any new Flow errors, it just reveals old ones that weren't noticed by Flow because of the code structure.

Reply:
> You can mark the presences as optional ala: `presences?: PresenceState`

I don't think we should do this since it'd be semantically incorrect: `presences` is not optional, it is always required. Furthermore, we always satisfy this requirement in `connect`, it's just not recognized by Flow. 

> Or maybe there is a better way to tell Flow the component is always getting the param from connect?

That'd be great. I think as long as we haven't found such a better way, we should leave this FlowFixMe as it is. (and perhaps open an issue in Flow / react-redux once we are certain this is an error on their side?

Reply:
> presences is not optional, it is always required. 
From the outside this property is optional (not even needed).

Reply:
I was thinking in terms of "`presence` will exist in every `ActivityText` object, therefore it shouldn't be marked optional in its Props". Otherwise, someone studying ActivityText.js  could wonder in what case `presences` would be optional, when this is actually never the case.

Reply:
Yeah, there's really two "props" types here: the type `this.props` will have inside the main component (we conventionally call this `Props`), and the type of the props that a caller needs to pass to the container which is the default export (we don't generally write this down explicitly).

Here, clearly `presences` is part of one of these types but not the other.

The thing that puzzles me looking at this is: it's very common for us to have some properties that are in the inner type but not the outer one! There's a difference here basically every time we use `connect` (or, previously, `connectWithActions`); sometimes we take the property from the caller if present, but even then that means the type of that prop should be optional on the outside and not on the inside. So: why aren't we running into this more often?

Reply:
Aha, I see -- it's that in that `connect` call, we claim the `props` parameter has type `Props`:
```
export default connect((state, props: Props) => ({
  presences: getPresence(state),
}))(ActivityText);
```

That's not right; but if we say so, then Flow correctly infers that the props passed to the container by its caller must have that type, and gives this error.

If I delete that annotation, the error goes away -- and also Flow successfully performs inference through it, so if I play with an `ActivityText` call site and introduce type errors, I get appropriate error messages.

So I'll just delete that annotation. It looks like that's the only place we make this mistake.

Reply:
We are using `?` more often :)

If we want to be both correct and precise, we would split the component into a `container` and a `dumb` component, then the container will not have this prop at all, and the `dumb` component will have this as `required` but it will also be clear that the component is passing props to it.

Another thing, in some Flow discussions people declare `own props` and `props from connect` as two different objects and then create a union between them.

Both of these are an improvement in but possibly too much boilerplate (or not?)